### PR TITLE
V 3/wui all wallet image

### PR DESCRIPTION
**Pull request by svenvoskamp**
# Changes

- feat: Add `wui-all-wallets-image` composite

### CODE REVIEW

File: apps/gallery/stories/composites/wui-all-wallets-image.stories.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import type { Meta } from '@storybook/web-components'
+import type {
+  WalletImage,
+  WuiAllWalletsImage
+} from '@web3modal/ui/src/composites/wui-all-wallets-image'
+import '@web3modal/ui/src/composites/wui-all-wallets-image'
+import '@web3modal/ui/src/composites/wui-wallet-image'
```
Comment (0xAsimetriq) [inline apps/gallery/stories/composites/wui-all-wallets-image.stories.ts:7]: Is standard wallet image import needed here?

File: apps/gallery/stories/composites/wui-all-wallets-image.stories.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      },
+      {
+        src: 'https://explorer-api.walletconnect.com/w3m/v1/getWalletImage/a7f416de-aa03-4c5e-3280-ab49269aef00?projectId=f63a70d1bd9dd78b5eb556c25d11cf05',
+        walletName: 'Ledger'
+      },
+      {
+        src: 'https://explorer-api.walletconnect.com/w3m/v1/getWalletImage/7e1514ba-932d-415d-1bdb-bccb6c2cbc00?projectId=f63a70d1bd9dd78b5eb556c25d11cf05',
+        walletName: 'Fireblocks'
+      }
+    ] as WalletImage[]
```
Comment (0xAsimetriq) [inline apps/gallery/stories/composites/wui-all-wallets-image.stories.ts:33]: I think this should be inherited from `as Component`? What was typescript saying here that make you add this assertion? Does `as const` work instead? if this works we don't need to export / import `WalletImage` type, as this is currently un-common along other components.

File: packages/ui/src/composites/wui-all-wallets-image/index.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  // -- state & properties ------------------------------------------- //
+  @property({ type: Array }) public walletImages: WalletImage[] = []
+
+  // -- render ------------------------------------------------------- //
+  public render() {
+    const walletImagesHtml = this.walletImages.map(walletImage =>
+      this.renderWalletImage(walletImage.src, walletImage.walletName)
+    )
+
+    return html` <div>${walletImagesHtml}</div> `
```
Comment (0xAsimetriq) [inline packages/ui/src/composites/wui-all-wallets-image/index.ts:24]: Do we need div here, perhaps we can style `:host` element directly? If so, this return can be simplified to

```
return this.walletImages.map(({src, alt})=> html`<wui-wallet-image size="inherit" src=${src} alt=${alt}></wui-wallet-image>`)
```

File: packages/ui/src/composites/wui-all-wallets-image/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  private renderWalletImage(src: string, alt: string) {
+    return html` <wui-wallet-image size="inherit" src=${src} alt=${alt}></wui-wallet-image> `
```
Comment (0xAsimetriq) [inline packages/ui/src/composites/wui-all-wallets-image/index.ts:28]:
```suggestion
    return html`<wui-wallet-image size="inherit" src=${src} alt=${alt}></wui-wallet-image>`
```

File: packages/ui/src/composites/wui-all-wallets-image/index.ts (hunk 3 above)
Comment (0xAsimetriq) [inline packages/ui/src/composites/wui-all-wallets-image/index.ts:24]:
```suggestion
    return html`<div>${walletImagesHtml}</div>`
```

**Review by 0xAsimetriq — CHANGES_REQUESTED** [review]

File: apps/gallery/stories/composites/wui-all-wallets-image.stories.ts (hunk 2 above)
Comment (svenvoskamp) [reply apps/gallery/stories/composites/wui-all-wallets-image.stories.ts]: Yes, you are right, it is inherited from `as Component` so not needed.

**Review by 0xAsimetriq — APPROVED** [review]

**Pull request merged**